prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -565,19 +562,23 @@ class FHIRRouter(
      * @param report Report object passed for logging purposes
      * @param receiver Receiver of the report
      * @param filterType Type of filter used
-     * @param focusResource Starting point for the evaluation, can add additional data to the log
+     * @param focusResource Starting point for the evaluation, logged if different than [bundle]
      */
     internal fun logFilterResults(
         filterName: String,
         bundle: Bundle,
         report: Report,
         receiver: Receiver,
         filterType: ReportStreamFilterType,
-        focusResource: Base? = null
+        focusResource: Base
     ) {
         var filteredTrackingElement = bundle.identifier.value ?: ""
-        if (focusResource != null) {
-            filteredTrackingElement += " at " + focusResource.idBase
+        if (focusResource is Observation) {
+            // for Observation-type elements, we use the code if property when available
+            // if more elements need specific logic, consider extending the FHIR libraries
+            // instead of adding more if/else statements
+            val coding = focusResource.code.coding.firstOrNull()
+            if (coding != null)  filteredTrackingElement += " with " + coding.system + " code:" + coding.code
```

Comment:
This works fine, although it takes out the idBase piece of it. I think the latest proposal was to use both the code and the id, would that still make sense?

Reply:
I think it makes sense either way actually, the id is mostly only useful for us to debug using the internal fhir, so I will defer to whatever you decide.